R/helpers.R

```diff
@@ -144,6 +144,31 @@ extract_num_iterations <- function(x) {
   length(evals)
 }
 
+# Extract weights for model features based on feature importance. Assumes that
+# the model was trained with the `valids` parameter set such that error metrics
+# are saved for each tree on the model$record_evals attribute. The output
+# weights are useful for computing comps using leaf node assignments
+extract_weights <- function(model, train, outcome_col, metric = "rmse") {
```

Comment:
This function is largely copied over from https://ccao-data.github.io/lightsnip/articles/finding-comps.html#quantifying-comparables with only a few tweaks: changing the argument order, removing unnecessary arguments, and getting rid of the `train()` call.